lib/spack/spack/cmd/pkg.py

```diff
@@ -180,17 +180,29 @@ def pkg_added(args):
         colify(sorted(u2))
 
 
-def pkg_changed(args):
-    """show packages changed since a commit"""
-    lower_type = args.type.lower()
+def get_all_package_diffs(type, rev1='HEAD^1', rev2='HEAD'):
```

Comment:
Can we move this function somewhere in "core" Spack, i.e. not as part of a `spack.cmd.`? In that way `spack.ci` doesn't need to depend on a `spack.cmd` package.

Reply:
Sure, any suggestions?  Otherwise, I'll look around and come up with a location.

Reply:
What about `lib/spack/spack/package.py`?  I would probably need to move some other methods in there as well, including `get_git`, and I'm not sure how good a fit that is.  There's also `lib/spack/spack/pkgkit.py` which currently seems to just collect stuff from other modules, but could maybe include a few module level methods.

Reply:
Oh, looking more, maybe not `pkgkit.py`, since it's goal is to make things available to package files.

Reply:
Definitely not pkgkit, package could make sense, or perhaps repo since it's an "all packages" action?

Reply:
I think we can put it under `spack.ci`? After all it tell, given 2 revisions, which packages changed in between. Where would that be useful if not in CI?

Reply:
In unit-test, style (which actually has another version of the same functionality already IIRC), probably other places too.

Reply:
Additionally, `cmd/pkg.py` already depends on `spack.repo`, so I kind of like it for that reason.

Reply:
Just wanted to note that unit-test and style are always run... in ci :laughing: Besides jokes the main point would be to avoid commands depending on "core" APIs and vice-versa so any core module mentioned here is fine with me for the time being.

Reply:
Ok, I had to move all the common methods, so they're living in `lib/spack/spack/repo.py` now.